### ISSUE TITLE
MonoTests.System.Runtime.Caching.MemoryCacheTest.Trim fails in Windows arm64

### ISSUE THREAD

**Issue by safern** [dotnet/runtime#36488]
### Builds
|Build|Pull Request | Test Failure Count|
| --- | --- | --- |
|[#636357](https://dev.azure.com/dnceng/public/_build/results?buildId=636357)|Rolling|1|
|[#644655](https://dev.azure.com/dnceng/public/_build/results?buildId=644655)|Rolling|1|
|[#645661](https://dev.azure.com/dnceng/public/_build/results?buildId=645661)|Rolling|1|
### Configurations
- net5.0-Windows_NT-Release-arm64-CoreCLR_release-Windows.10.Arm64.Open
### Helix Logs
|Build|Pull Request|Console|Core|Test Results|Run Client|
| --- | --- | --- | --- | --- | --- |
|[#636357](https://dev.azure.com/dnceng/public/_build/results?buildId=636357)|Rolling|[console.log](https://helix.dot.net/api/2019-06-17/jobs/6eefa3b0-17e2-40c0-927c-5c76c2171dcd/workitems/System.Runtime.Caching.Tests/files/console.ec284435.log)||[testResults.xml](https://helix.dot.net/api/2019-06-17/jobs/6eefa3b0-17e2-40c0-927c-5c76c2171dcd/workitems/System.Runtime.Caching.Tests/files/testResults.xml)|[run_client.py](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-master-6eefa3b017e240c092/System.Runtime.Caching.Tests/1874db06-1356-4542-9935-ab75e601fc87.log?sv=2019-02-02&se=2020-05-28T08%3A11%3A15Z&sr=c&sp=rl&sig=GFVWQxAoFB06Nbk%2BOcUFgUxdhNgHKaZ5jOpDu5mgi1w%3D)|
|[#644655](https://dev.azure.com/dnceng/public/_build/results?buildId=644655)|Rolling|[console.log](https://helix.dot.net/api/2019-06-17/jobs/dc98fb0f-5f6a-41cb-af69-40512f2be70e/workitems/System.Runtime.Caching.Tests/files/console.5c439e3f.log)||[testResults.xml](https://helix.dot.net/api/2019-06-17/jobs/dc98fb0f-5f6a-41cb-af69-40512f2be70e/workitems/System.Runtime.Caching.Tests/files/testResults.xml)|[run_client.py](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-master-dc98fb0f5f6a41cbaf/System.Runtime.Caching.Tests/da846051-d947-49ed-89d2-ae40bb1a400f.log?sv=2019-02-02&se=2020-06-03T13%3A19%3A05Z&sr=c&sp=rl&sig=e3kwuEyK9MpeS06LNsImCx%2F3lAiPIvG1EIzAJCdIBrQ%3D)|
|[#645661](https://dev.azure.com/dnceng/public/_build/results?buildId=645661)|Rolling|[console.log](https://helix.dot.net/api/2019-06-17/jobs/062e5613-0d24-4a50-b8ce-6c02d25f54a6/workitems/System.Runtime.Caching.Tests/files/console.c90bcb86.log)||[testResults.xml](https://helix.dot.net/api/2019-06-17/jobs/062e5613-0d24-4a50-b8ce-6c02d25f54a6/workitems/System.Runtime.Caching.Tests/files/testResults.xml)|[run_client.py](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-master-062e56130d244a50b8/System.Runtime.Caching.Tests/c43a10b3-ef3a-4ca6-83c4-b64b229b1e3f.log?sv=2019-02-02&se=2020-06-03T22%3A52%3A10Z&sr=c&sp=rl&sig=DlVkCkcLpYT%2FIGfAaLoUWDU7JgO7Yk46A0A2NJ%2F3htQ%3D)|

```
testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing  
  Discovering: System.Runtime.Caching.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Runtime.Caching.Tests (found 29 of 36 test cases)
  Starting:    System.Runtime.Caching.Tests (parallel test collections = on, max threads = 46)
    MonoTests.System.Runtime.Caching.MemoryCacheTest.Trim [FAIL]
      Assert.Equal() Failure
      Expected: 23
      Actual:   22
      Stack Trace:
        /_/src/libraries/System.Runtime.Caching/tests/System.Runtime.Caching/MemoryCacheTest.cs(992,0): at MonoTests.System.Runtime.Caching.MemoryCacheTest.Trim()
  Finished:    System.Runtime.Caching.Tests
=== TEST EXECUTION SUMMARY ===
   System.Runtime.Caching.Tests  Total: 29, Errors: 0, Failed: 1, Skipped: 0, Time: 4.611s
```
```

**Comment by KKhurin** [comment]
I understand why we failed here. One of the cache stores got 4 items (instead of expected 1) so trimming 75% of it resulted in 1 item surviving the trim. To greatly reduce the chance of seeing it again we can trim 76%...

**Comment by safern** [comment]
@KKhurin just to update here I ran the test without the fix in a loop 30 times and it failed 3 times. 

I ran the test with the fix (changing to 76%) in a loop 50 times and it didn't fail, so I think it makes sense to put that fix. I will submit a PR. Thanks for the suggestion.

Run without the fix: https://helix.dot.net/api/2019-06-17/jobs/b7f84fb0-3804-49f6-97dc-d09fc2180585/workitems/NoFix/console

Run with the fix: https://helix.dot.net/api/2019-06-17/jobs/bc764363-3621-4740-a24c-d336546cb6c2/workitems/Fix/console
